storops/unity/resource/host.py

```diff
@@ -74,6 +76,80 @@ def get_host(cls, cli, _id, force_create=False):
             ret = cls.get(cli=cli, _id=_id)
         return ret
 
+    def detach_hlu(self, lun):
+        return lun.detach_from(self)
+
+    def has_hlu(self, lun):
+        if not self.host_luns or not lun:
+            return False
+
+        has = False
+        for host_lun in self.host_luns:
```

Comment:
These lines are duplicated.
I would suggest to create a property like this:

```python
@properties
@instance_cache
def lun_ids(self):
    ret = []
    if self.host_luns:
        for host_lun in self.host_luns:
            if host_lun.lun is not None:
                ret.append(host_lun.lun.id)
    return ret

def has_hlu(self, lun):
    return lun.id in self.lun_ids
```

The `@instance_cache` here works as an instance level memoize.